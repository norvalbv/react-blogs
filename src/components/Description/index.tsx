import React, { ReactElement } from 'react';
import { DefTheme } from 'types';
import { getClassName, isLabelInProps } from 'utils';

export type DescriptionProps = {
  children?: string | ReactElement;
  className?: string;
  props?: unknown;
  theme?: DefTheme;
};

const Description = ({
  children,
  className,
  theme,
  ...props
}: DescriptionProps): ReactElement | null => {
  if (!children) {
    return null;
  }

  if (typeof children === 'string') {
    return (
      <p className={getClassName({ tag: 'h1', theme, className })} {...isLabelInProps(props)}>
        {children}
      </p>
    );
  }

  return children;
};

export default Description;
