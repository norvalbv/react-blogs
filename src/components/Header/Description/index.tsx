import React, { ReactElement } from 'react';

export type DescriptionProps = {
  children?: string | JSX.Element;
  className?: string;
  testId?: string;
};

const Description = ({ children, className, testId }: DescriptionProps): ReactElement | null => {
  if (!children) {
    return null;
  }

  if (typeof children === 'string') {
    return (
      <p className={className} data-testid={testId || `${children} Description`}>
        {children}
      </p>
    );
  }

  return children;
};

export default Description;
