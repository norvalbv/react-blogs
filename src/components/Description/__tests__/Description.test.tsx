import React, { FC } from 'react';
import renderer from 'react-test-renderer';

import Description, { DescriptionProps } from '..';

const DescriptionWithAllProps: FC<Required<DescriptionProps>> = Description;

describe('<Subtitle />', () => {
  test('should render nothing if subtitle is not provided', () => {
    const tree = renderer.create(<Description />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render correctly with all props', () => {
    const tree = renderer
      .create(
        <DescriptionWithAllProps
          className="text-sm"
          props={{ testId: 'Test Id' }}
          theme={{
            theme: undefined,
            code: undefined,
            overrides: undefined,
          }}
        >
          Test subtitle
        </DescriptionWithAllProps>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
