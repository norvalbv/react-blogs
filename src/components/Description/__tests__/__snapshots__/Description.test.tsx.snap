// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Subtitle /> should render correctly with all props 1`] = `
<p
  props={
    {
      "testId": "Test Id",
    }
  }
>
  Test subtitle
</p>
`;

exports[`<Subtitle /> should render nothing if subtitle is not provided 1`] = `null`;
